### ISSUE TITLE
AI command doesn't capture text

### ISSUE THREAD

**Issue by neominik** [BoltAI/BoltAI#37]
When I try to use the AI command when I have highlighted some text in any app, instead of correctly pasting the text into triple quotes, the propmt always results in the following:
```
My notes are below in triple quotes:

""" PICTURE """

Write a markdown list (using dashes) of key takeaways from my notes. Write at least 10 items. Do not wrap responses in quotes.
```
"PICTURE" is not the highlighted text, obviously. Unfortunately, this makes AI commands unusable, unless I copy the text and paste it when editing the message, at which point I don't need a command for this.
Accessibility privileges have been granted, version 1.14.1, macOS 14.3.1

**Comment by neominik** [comment]
Update: After restarting macOS, I am now getting an error popup saying "Failed to get the highlighted text"
Any idea how to fix this?

**Comment by longseespace** [comment]
Hi. Can you add more information on the app you were trying to use an AI Command?

Make sure that app is in active. In many cases the text is highlighted but the currently active app was BoltAI and not that third party app and it caused the issue.

Another possible reason is BoltAI failed to use the Accessibility permissions. 

Can you delete the entry in System Settings > Accessibility and add BoltAI to that list again

Thanks

**Comment by neominik** [comment]
I tried different apps: Slack, Arc, Safari, but all apps gave the same result. I also deleted and re-added the entry in the accessibility permissions, but it didn't change the error popup. However, today without changing anything, it is back to not working in a different way. I figured out that it will paste whatever text is in the clipboard into the triple quotes. Presumably without properly copying from the active app before.
Is it supposed to copy and paste? If so, does it try to send the Cmd+C keystroke to the app via some keycode? If that is the case, it might explain why it isn't working, because I'm using a different keyboard layout.

**Comment by longseespace** [comment]
Ah yes, you're right about copying and paste.

This problem is quite tricky. BoltAI uses the standard (albeit unpopular) API called Accessibility API. For standard app with Cocoa UI components, it can get the text content properly.

But for many web-based apps with custom UI components, it simply doesn't work.

And so one "hack" is to copy the text to clipboard instead. That I figure that might be the issue in your case.

I will try to find a better solution for this.

**Comment by neominik** [comment]
This seems to have been resolved in v1.21.0 as well 👍🏻
